openpype/pipeline/publish/lib.py

```diff
@@ -864,6 +864,109 @@ def _validate_transient_template(project_name, template_name, anatomy):
                          ).format(template_name, project_name))
 
 
+def get_published_workfile_instance(context):
+    """Find workfile instance in context"""
+    for i in context:
+        is_workfile = (
+            "workfile" in i.data.get("families", []) or
+            i.data["family"] == "workfile"
+        )
+        if not is_workfile:
+            continue
+
+        # test if there is instance of workfile waiting
+        # to be published.
+        if not i.data.get("publish", True):
+            continue
+
+        return i
+
+
+def replace_published_scene(instance, replace_in_path=True):
+    """Switch work scene for published scene.
+    If rendering/exporting from published scenes is enabled, this will
+    replace paths from working scene to published scene.
```

Comment:
Let's have the docstring also state that it only works if the current publish context is also publishing an enabled "workfile" instance.